aptos-move/framework/src/natives/cryptography/algebra/eq.rs

```diff
@@ -0,0 +1,75 @@
+// Copyright © Aptos Foundation
+
+use crate::{
+    abort_unless_arithmetics_enabled_for_structure, abort_unless_feature_flag_enabled,
+    natives::{
+        cryptography::algebra::{
+            abort_invariant_violated, feature_flag_from_structure, gas::GasParameters,
+            AlgebraContext, Structure, MOVE_ABORT_CODE_NOT_IMPLEMENTED,
+        },
+        helpers::{SafeNativeContext, SafeNativeError, SafeNativeResult},
+    },
+    safe_borrow_element, safely_pop_arg, structure_from_ty_arg,
+};
+use move_core_types::gas_algebra::NumArgs;
+use move_vm_types::{loaded_data::runtime_types::Type, values::Value};
+use smallvec::{smallvec, SmallVec};
+use std::collections::VecDeque;
+
+macro_rules! ark_eq_internal {
```

Comment:
Should this be also moved to `aptos-move/framework/src/natives/cryptography/algebra/arithmetics/mod.rs`?